cohortextractor/cohortextractor.py

```diff
@@ -817,18 +859,37 @@ def main():
                 "generate_cohort: error: one of the arguments "
                 "--expectations-population --dummy-data-file --database-url is required"
             )
+        if options.output_file:
+            if options.output_dir or options.output_format:
+                parser.error(
+                    "generate_cohort: error: cannot combine --output-file argument "
+                    "with --output-dir or --output-format"
+                )
+            output_dir = options.output_file.parent
+            match = re.match(
+                rf"^(.+)\.({EXTENSION_REGEX})$", str(options.output_file.name)
+            )
+            output_name = match.group(1)
+            output_format = match.group(2)
```

Comment:
What happens if a `--output-file` has an unsupported extension?  Or some typo that means it doesn't match?  Does it need to be caught here to give a nicer error message than an error in the regex-match?

Reply:
Good point, I've added code to catch this now.